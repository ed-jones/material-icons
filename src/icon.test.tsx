import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/react';

import Outlined from './outlined';
import Rounded from './rounded';
import Sharp from './sharp';

describe('Icon', () => {
	it('should render outlined', () => {
		const { asFragment } = render(<Outlined icon='folder' />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render rounded', () => {
		const { asFragment } = render(<Rounded icon='folder' />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render sharp', () => {
		const { asFragment } = render(<Sharp icon='folder' />);
		expect(asFragment()).toMatchSnapshot();
	});

	it('should render with props', () => {
		const { asFragment } = render(
			<Sharp icon='folder' fill weight={100} grade={4} size={32} color='red' as='button' />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
