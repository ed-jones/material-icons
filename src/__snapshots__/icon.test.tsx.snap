// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Icon > should render icon 1`] = `
<DocumentFragment>
  <span
    class="material-symbols"
  >
    folder
  </span>
</DocumentFragment>
`;

exports[`Icon > should render with props 1`] = `
<DocumentFragment>
  <button
    class="material-symbols"
    style="color: red; font-variation-settings: \\"FILL\\" 1, \\"wght\\" 100, \\"GRAD\\" 4, \\"opsz\\" 32; font-size: 32px;"
  >
    folder
  </button>
</DocumentFragment>
`;
